std/conv.d

```diff
@@ -6136,6 +6140,88 @@ private auto hexStrLiteral(String)(scope String hexData)
     assert(hexString!"ab cd" == hexString!"ABCD");
 }
 
+/************************************************
+ * Convert string of `[0-9A-Fa-f]` characters to an array of ubytes.
+ *
+ * Other characters in the string are considered separators.
+ * Adjacent pairs of characters form the nibbles of the ubyte.
+ * Non-pairing characters form the low nibble of the ubyte.
+ * The returned array does not have a terminating zero appended.
+ * The GC is used because `hexData()` is intended to be used
+ * at compile time to, for example, initialize manifest constants
+ * and statically initialize global variables.
+ * Params:
+ *      s = string of data to convert
+ * Returns:
+ *      unique immutable array of ubytes allocated on the GC heap
+ * References:
+ *      $(LREF hexString)
+ */
+immutable(ubyte)[] hexData(const(char)[] s) pure nothrow @trusted
```

Comment:
Since the function is `pure`, it doesn't *have* to be immutable, right? Making the return type mutable (`ubyte[]`) still allows initializing immutable data because it's `pure`.

https://run.dlang.io/is/eH2zDC

Reply:
you're right